processing/data_collection/gazette/spiders/pr_curitiba.py

```diff
@@ -0,0 +1,117 @@
+from dateparser import parse
+import datetime as dt
+import re
+
+from gazette.spiders.base import BaseGazetteSpider
+
+import scrapy
+
+from gazette.items import Gazette
+
+
+class PrCuritibaSpider(BaseGazetteSpider):
+    TERRITORY_ID = '4106902'
+    name = 'pr_curitiba'
+    allowed_domains = ['legisladocexterno.curitiba.pr.gov.br']
+    custom_settings = {
+        'DEFAULT_REQUEST_HEADERS': {
+            'user-agent': 'Mozilla/5.0'
+        }
+    }
+
+    def start_requests(self):
+        """
+        The Curitiba website is a statefull page, so we can't just build the
+        request from zero, we have to resend the viewstate with every request.
+        @url http://legisladocexterno.curitiba.pr.gov.br/DiarioConsultaExterna_Pesquisa.aspx
+        @returns requests 1
+        """
+        todays_date = dt.date.today()
+        current_year = todays_date.year
+        for year in range(current_year, 2006, -1):
+            yield scrapy.FormRequest(
+                'http://legisladocexterno.curitiba.pr.gov.br/DiarioConsultaExterna_Pesquisa.aspx',
+                formdata={
+                    'ctl00$cphMasterPrincipal$ddlGrAno': str(year)
+                },
+                callback=self.parse_year
+            )
+
+    def parse_year(self, response):
+        for i in range(12):
```

Comment:
`i` is not a good variable name. It seems you are trying to search for months, right? Maybe `month` is a better name.
Anyway, this will fail in future months (for example, considering the date of this review, december/2018). Maybe include some validation to avoid trying to get future months.